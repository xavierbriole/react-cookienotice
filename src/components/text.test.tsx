import { render } from '@testing-library/react'

import Text from './text'

describe('Text', () => {
  describe('should render', () => {
    it('with className', () => {
      const { asFragment } = render(<Text className='className'>children</Text>)

      expect(asFragment()).toMatchSnapshot()
    })

    it('without className', () => {
      const { asFragment } = render(<Text className={undefined}>children</Text>)

      expect(asFragment()).toMatchSnapshot()
    })
  })

  it('should display text', () => {
    const { getByText } = render(<Text>children</Text>)

    expect(getByText('children')).toBeInTheDocument()
  })
})
