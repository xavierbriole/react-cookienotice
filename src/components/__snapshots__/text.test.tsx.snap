// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Text > should render > with className 1`] = `
<DocumentFragment>
  <div
    class="className"
  >
    <span>
      children
    </span>
  </div>
</DocumentFragment>
`;

exports[`Text > should render > without className 1`] = `
<DocumentFragment>
  <div>
    <span>
      children
    </span>
  </div>
</DocumentFragment>
`;
